### ISSUE TITLE
RC4 encrypt/decrypt in radare2

### ISSUE THREAD

**Issue by Maijin** [radare/radare2#4229]
Something really simple like this:

```
with open("/path/to/file.txt", "r") as encrypted_file:
    data = encrypted_file.read()
key = "<rc4 key>"

S = range(256)
j = 0
out = []

#KSA Phase
for i in range(256):
    j = (j + S[i] + ord( key[i % len(key)] )) % 256
    S[i] , S[j] = S[j] , S[i]

#PRGA Phase
i = j = 0
for char in data:
    i = ( i + 1 ) % 256
    j = ( j + S[i] ) % 256
    S[i] , S[j] = S[j] , S[i]
    out.append(chr(ord(char) ^ S[(S[i] + S[j]) % 256]))

decrypted_text = ''.join(out)
with open('decrypted.txt', 'w') as decrypted_file:
    decrypted_file.write(decrypted_text)
```

Would be definitely super useful for malware analysis ! It can also be done as an r2pm module using the script as it is !

**Comment by radare** [comment]
Rc4 is quite simple so it can be added in rhash/rcrypto api

**Comment by radare** [comment]
```
$ rahash2 -S key -E rc4 -s hello' world\n' | rahash2 -S key -E rc4 -s -
hello world
```

**Comment by radare** [comment]
Fixed in 3f5aa46bd5177f18a6e784c45e0ec3da9c1eb266
